### PR TITLE
Make directory domain nullable

### PR DESCRIPTION
**Pull request by blairlunceford**
Makes directory `domain` parameter nullable, as we're pulling this from the organization's domains which can be empty.

### CODE REVIEW

**Comment by marktran** [comment]
@mthadley Can you take a look? 👀

File: spec/support/fixtures/vcr_cassettes/directory_sync/list_directories/with_after.yml [hunk 1]
```diff
@@ -52,24 +52,36 @@ http_interactions:
...
-        Test","external_key":"rPzV4pdpbaUiKsc6","type":"bamboohr","state":"unlinked","domain":"foo-corp.com", "created_at":"2021-07-02T19:15:39.556Z","updated_at":"2021-07-02T19:31:28.499Z"},{"object":"directory","id":"directory_01F5ZY7XVQZ3DRYEZTH1EPA8BS","organization_id":"org_01EZDF20TZEJXKPSX2BJRN6TV6","name":"Foo
-        Corp","external_key":"qV4eyK99QGUaYYa0","type":"okta scim v2.0","state":"linked","domain":"foo-corp.com", "created_at":"2021-07-02T19:15:39.556Z","updated_at":"2021-07-02T19:31:28.499Z"},{"object":"directory","id":"directory_01F5XHH1QHX6C2F0Z6WG9YPGCJ","organization_id":"org_01F29YJ068E52HGEB8ZQGC9MJG","name":"Example
-        Azure SCIM","external_key":"YDKJvbWHKKg66cSk","type":"azure scim v2.0","state":"linked","domain":"example.com", "created_at":"2021-07-02T19:15:39.556Z","updated_at":"2021-07-02T19:31:28.499Z"}]}'
+      encoding: ASCII-8BIT
+      string: '{"object":"list","data":[{"object":"directory","id":"directory_01FK3HFFGMC2WF32RR8SKWC8KA","organization_id":"org_01FHRF60X0CDBB8F6A0YGSJJ2B","name":"Azure
```
Comment (mthadley) [inline spec/support/fixtures/vcr_cassettes/directory_sync/list_directories/with_after.yml:80]: Looks like a few of these directories don't have a `domain`, so I think we have coverage of that case now. 👍

**Review by mthadley — APPROVED** [review]
Looks good!

Not sure why CI is sporadically failing on different Ruby versions, with `post_install hook failed for workos-1.6.0`.

**Comment by blairlunceford** [comment]
> Looks good!
> 
> Not sure why CI is sporadically failing on different Ruby versions, with `post_install hook failed for workos-1.6.0`.

Odd - it's passing now: https://workos.semaphoreci.com/branches/009c571e-3b19-4e45-aa49-05dc328f1515 but I'm not seeing the success message in this PR.

**Pull request merged**